packages/mui-material-next/migration.md

```diff
@@ -6,6 +6,58 @@ This is a reference guide on how to migrate from Material UI v5 to v6.
 
 This section lists all breaking changes related to components in v6 and how to address them.
 
+## Overarching changes
+
+These are the changes that apply to all components
+
+### Remove `components` and `componentsProps` props
+
+The deprecated `components` and `componentsProps` props are removed in v6.
+If you were using these, then you can use `slots` and `slotProps` props instead,
+which have the same functionality and API.
+Here's an example of the change using the Badge component:
+
+```diff
+ <Badge
+     badgeContent={4}
+-    components={{ badge: (props) => <span {...props}>slot test</span> }}
++    slots={{ badge: (props) => <span {...props}>slot test</span> }}
+-    componentsProps={{ badge: { id: "test-id" } }}
++    slotProps={{ badge: { id: "test-id" } }}
+ >
+   <MailIcon color="action" />
+ </Badge>
+```
+
+## Badge
+
+### Added and removed colors
+
+Color options were changed to match Material You:
+
+- `"default"`, `"info"`, `"success"`, and `"warning"` were removed and are no longer supported out-of-the-box
```

Comment:
Should we still keep these? We dsicussed that it may be useful to define these in the team. cc @danilo-leal

Reply:
I discussed with @danilo-leal and we will work on a Material UI Material You theme that makes upgrading smoother (for example, having these colors) and that makes sense with the Material You color system.

I'll be working on developing this, but there are still things to figure out, like for example which would be the default theme, the Material You default theme, or the Material UI MY theme.

I don't want to block the components progress while we figure out the Material UI MY theme, so for now I think we should continue the work using the Material You default theme (which doesn't have these colors) and when we have more clarity we can go back to this. What do you think?

Reply:
My worry with this is that we would introduce a breaking change and later add back these values. It could be frustrating for people to have to do some changes, and revert them later.

Reply:
You're right, what we should do then is add the `"info"`, `"success"`, and `"warning"` to the palette following Material You's custom colors guidelines: https://m3.material.io/styles/color/the-color-system/custom-colors.

Regarding the `"default"` color, that's not really a color but rather the badge has no background:

<img width="59" alt="Screenshot 2023-07-21 at 08 44 32" src="https://github.com/mui/material-ui/assets/16889233/12790971-a9d5-4950-872e-1d4dc39e30d7">

Do you think we should keep that one as well?

I'll work with @zanivan on this.

Reply:
In v5 we started with removing "default" from the color values, I wonder why it is still there in the `Badge`. The reasoning for it was that `default` is not a color, we should use as default some valid color value. I propose we drop it.